src/hooks/use-products.jsx

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react';
+import { getProducts, getCategorizedProducts } from './../data/API';
+import { useParams } from 'react-router-dom';
+
+export default function useProducts(sort) {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
+  const [products, setProducts] = useState([]);
+  const filters = ['정확도순', '낮은 가격순', '높은 가격순'];
+  const [filter, setFilter] = useState(filters[0]);
+  const filtered = getFilteredProducts(filter, products);
+  const { tag } = useParams();
+  const searchText = '';
+  const searchTags = [tag];
+  const [category, setCategory] = useState([]);
+  const [select, setSelect] = useState(filter[0]);
+
+  useEffect(() => {
+    setLoading(true);
+    setError(undefined);
+    let productsData;
+    sort === 'total'
+      ? (productsData = getProducts())
+      : tag === '전체상품' || tag === '베스트'
+      ? (productsData = getProducts())
+      : (productsData = getCategorizedProducts(searchText, searchTags));
+    productsData
+      .then((data) => {
+        // console.log('fetching...');
+        setProducts(data);
+      })
+      .catch((e) => setError('에러가 발생했습니다'))
+      .finally(() => setLoading(false));
```

Comment:
```suggestion
    productsData
      .then((data) => {
        // console.log('fetching...');
        setProducts(data);
      })
      .catch((error) => setError(`에러가 발생했습니다: ${error.message} `))
      .finally(() => setLoading(false));
```
error 는 e 로 축약하지 않기로 해요 [컨벤션](https://www.notion.so/hnhyerim/code-updated-12-22-f4f8aa0f6de9424ab2cad142df625d83#3a9a7b61ea1544f4b1ff6624c40e2894)
또한 error 가 발생여부 보다는, 
에러의 구체적인 원인을 알려주고 수정하게끔 해당 정보를 개발자가 볼수 있어야 합니다.
catch 메소드가 error 객체를 전달해주므로 그 객체에 있는 정보를 활용하는 게 좋습니다[[Error]](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Error)